packages/clerk-js/src/ui/components/UserProfile/Web3Section.tsx

```diff
@@ -28,75 +28,86 @@ const shortenWeb3Address = (address: string) => {
   return address.slice(0, 6) + '...' + address.slice(-4);
 };
 
-export const Web3Section = withCardStateProvider(() => {
-  const { user } = useUser();
-  const card = useCardState();
-  const { strategyToDisplayData } = useEnabledThirdPartyProviders();
+export const Web3Section = withCardStateProvider(
+  ({ shouldAllowCreation = true }: { shouldAllowCreation?: boolean }) => {
+    const { user } = useUser();
+    const card = useCardState();
+    const { strategyToDisplayData } = useEnabledThirdPartyProviders();
 
-  return (
-    <ProfileSection.Root
-      title={localizationKeys('userProfile.start.web3WalletsSection.title')}
-      id='web3Wallets'
-    >
-      <Card.Alert>{card.error}</Card.Alert>
-      <Action.Root>
-        <ProfileSection.ItemList id='web3Wallets'>
-          {user?.web3Wallets.map(wallet => {
-            const strategy = wallet.verification.strategy as keyof typeof strategyToDisplayData;
+    /**
+     * Does not render if:
+     * - User is not present
+     * - Creation isn't allowed AND there aren't external accounts
+     */
+    if (!user || (!shouldAllowCreation && user.web3Wallets.length === 0)) {
```

Comment:
```suggestion
    if (!shouldAllowCreation && user.web3Wallets.length === 0) {
```

Not sure if we should check for `!user` here since we didn't have this check previously 🤔 AFAIK it's an impossible case to have `user` undefined within that component.